File: benchmark/php/collatz.php
<?php
$start = hrtime(true);
$max_steps = 0; $max_n = 0;
for ($n = 1; $n <= 10000; $n++) {
    $steps = 0; $c = $n;
    while ($c != 1) {
        if ($c % 2 == 0) { $c = intdiv($c, 2); }
        else { $c = $c * 3 + 1; }
        $steps++;
    }
    if ($steps > $max_steps) { $max_steps = $steps; $max_n = $n; }
}
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Max steps: " . $max_steps . " at n=" . $max_n . "
";
echo "Time: " . round($ms) . "ms
";

File: benchmark/ruby/collatz.rb
start = Time.now.to_f * 1000
max_steps = 0
max_n = 0
(1..10000).each do |n|
  steps = 0
  current = n
  while current != 1
    if current % 2 == 0
      current = current / 2
    else
      current = current * 3 + 1
    end
    steps += 1
  end
  if steps > max_steps
    max_steps = steps
    max_n = n
  end
end
finish = Time.now.to_f * 1000
puts "Max steps: #{max_steps} at n=#{max_n}"
puts "Time: #{(finish - start).round}ms"

